File: exercises/floor01/ex02_strings.py
"""
FLOOR 1 · EXERCISE 1-2 — Strings & f-strings
Lesson: 1.2

Every roguelike is a machine that prints strings. HUD lines, damage popups,
death screens — all strings. f-strings are how Python builds them cleanly.

Run:  python3 dojo.py check 1-2
Save: python3 dojo.py done 1-2
"""

player_name = "Crumb"
player_hp = 24
player_max_hp = 30
depth = 2

# --- Part 1: f-string HUD ----------------------------------------------------
# Build EXACTLY this string using an f-string and the variables above
# (don't type the numbers by hand — the HUD must update when the vars do):
#
#   Crumb | HP 24/30 | Floor 2
#
# TODO:
hud_line = ...

# --- Part 2: methods ---------------------------------------------------------
# The dungeon announcer SHOUTS. Take the quiet message below and produce
# the loud version using a string method (hint: .upper()).

quiet = "a wild rat approaches"
# TODO:
loud = ...

# --- Part 3: spot the trap ---------------------------------------------------
# Someone tried to build a damage popup by adding a number to a string and
# the game crashed (TypeError: can only concatenate str to str).
# Fix it with an f-string instead of the broken + version.

damage = 7
# broken:  popup = "You take " + damage + " damage!"
# TODO — same text, but working:
popup = ...

File: exercises/cpp/floor01/ex02_strings.cpp
// FLOOR C1 · EXERCISE c1-2 — std::string & std::to_string
// Lesson: c1.2
//
// C++ has no f-strings. Strings are built with + and std::to_string(number).
// Same HUD as the Python wing — feel the difference in the hands.
//
// Run:  python3 dojo.py check c1-2
// Save: python3 dojo.py done c1-2

#include <string>

std::string player_name = "Crumb";
int player_hp = 24;
int player_max_hp = 30;
int depth = 2;

// --- Part 1: the HUD ----------------------------------------------------------
// Build EXACTLY:   Crumb | HP 24/30 | Floor 2
// from the variables above. Numbers must pass through std::to_string —
// gluing an int onto a string directly won't compile (try it, read the error).
std::string make_hud() {
    // TODO
    return "";
}

// --- Part 2: shouting ------------------------------------------------------------
// Uppercase the message. C++ has no .upper() on strings (!) — loop over the
// characters and use toupper from <cctype> on each. Yes, really. Welcome to
// C++: closer to the metal, batteries not included.
#include <cctype>
std::string shout(std::string msg) {
    // TODO: for (char& c : msg) { ... }   then return msg
    return "";
}

// --- Part 3: the popup ------------------------------------------------------------
// Return: You take 7 damage!   (for damage=7 — use the parameter, not a literal 7)
std::string damage_popup(int damage) {
    // TODO
    return "";
}
